strawberryfields/parameters.py

```diff
@@ -350,6 +350,13 @@ def _eval_evalf(self, prec):
                     self
                 )
             )
+
+        # remove unnecessary dims when returning measurement
+        try:
+            res = np.squeeze(res).item()
+        except ValueError:
+            res = np.squeeze(res)
```

Comment:
As far as I can tell, this will only affect the case where measured registers are used in subsequent ops:

```python
MeasureX | q[0]
Xgate(q[0].par) | q[1]
```

I don't _think_ this affects the TF backend in any way, since the above is not differentiable, but just wanted to double check.

Reply:
I was thinking the same. Your example would be failing earlier, since e.g. the `Xgate` wouldn't accept a nested `ndarray` as input, which is the reason for this part of the code; to turn `np.array([[3]])` and `tf.Tensor([[3]])` into `3`, and `np.array([[1, 2]])`  and `tf.Tensor([[1, 2]])` into `np.array([1, 2])`, etc.

Reply:
When using the TensorFlow backend, will `res` be a TensorFlow tensor? e.g.:

```python
prog = sf.Program(2)

with prog.context as q:
    MeasureX | q[0]
    Xgate(q[0].par) | q[1]

eng = sf.Engine("tf", backend_options={"cutoff_dim":5})
eng.run(prog)
```

Will this fail because `np.squeeze` won't know how to deal with a Tensor?

Reply:
I don't think it will, since `np.squeeze` will change the `tf.Tensor` into a `np.ndarray` and then "squeeze" it. At least that's how it seems to work. Your code-snippet works fine locally for me.

Reply:
Awesome